### ISSUE TITLE
Folia Support for DecentHolograms?

### ISSUE THREAD

**Issue by Agaloth** [DecentSoftware-eu/DecentHolograms#123]
### Just making sure

- [X] I have read the wiki and made sure, this feature is not yet in the plugin.
- [X] I believe, that this feature is possible to implement.
- [X] I made sure, that this feature has not been requested yet.

### Problem

There is no Folia Support for DecentHolograms yet (since it just came out)

### Feature

https://github.com/PaperMC/Folia

Here is a summary of everything made with ChatGPT to save some time:

__For plugin developers, there are a few key takeaways from the given text:__

1. Folia introduces regionised multithreading, which means that plugins will require modification to function properly. Plugin developers should expect compatibility with existing plugins to be at zero.

2. Folia introduces new API additions, such as the RegionisedScheduler and EntityScheduler, which can help plugins run on the correct thread context. These additions are not compatible with Paper, but there are plans to add API to allow Folia plugins to be compatible with Paper.

3. To ensure compatibility with Folia, plugin developers must mark their plugin as compatible by placing "folia-supported: true" in the plugin.yml file. Additionally, plugins must take advantage of the RegionisedScheduler and EntityScheduler API to ensure correct behavior.

4. Plugins should not assume that they can access or modify data in other regions, as Folia regions tick in parallel and do not share data. Plugin developers should take advantage of the upcoming thread-check API to ensure correct behavior.

5. There are currently broken API, including the scoreboard API, and planned API changes, such as super aggressive thread checks, which may impact plugin development.

Overall, plugin developers should be prepared for significant changes in their plugins to ensure compatibility with Folia. They should take advantage of the new API additions and mark their plugin as compatible to ensure correct behavior. Additionally, they should be aware of the limitations and rules of Folia's regionised multithreading to avoid bugs and errors in their plugins.

**Comment by Aitooor** [comment]
Will it this possible?

**Comment by cgw2023** [comment]
> Try: https://github.com/mani1232/DecentHolograms

Is there any jar can download for testing?

**Comment by mani1232** [comment]
> > Try: https://github.com/mani1232/DecentHolograms
> 
> Is there any jar can download for testing?

[DecentHolograms-2.8.3-all.zip](https://github.com/DecentSoftware-eu/DecentHolograms/files/11862594/DecentHolograms-2.8.3-all.zip)

**Comment by LuluxClient** [comment]
Hey any updates on that ?

**Comment by cgw2023** [comment]
> > > Try: https://github.com/mani1232/DecentHolograms
> > 
> > 
> > Is there any jar can download for testing?
> 
> [DecentHolograms-2.8.3-all.zip](https://github.com/DecentSoftware-eu/DecentHolograms/files/11862594/DecentHolograms-2.8.3-all.zip)

is there a 1.20.2 version?

**Comment by mani1232** [comment]
> > > > Try: https://github.com/mani1232/DecentHolograms
> > > 
> > > 
> > > Is there any jar can download for testing?
> > 
> > 
> > [DecentHolograms-2.8.3-all.zip](https://github.com/DecentSoftware-eu/DecentHolograms/files/11862594/DecentHolograms-2.8.3-all.zip)
> 
> is there a 1.20.2 version?

Compile https://github.com/mani1232/DecentHolograms for latest version

**Comment by cgw2023** [comment]
> > > > > Try: https://github.com/mani1232/DecentHolograms
> > > > 
> > > > 
> > > > Is there any jar can download for testing?
> > > 
> > > 
> > > [DecentHolograms-2.8.3-all.zip](https://github.com/DecentSoftware-eu/DecentHolograms/files/11862594/DecentHolograms-2.8.3-all.zip)
> > 
> > 
> > is there a 1.20.2 version?
> 
> Compile https://github.com/mani1232/DecentHolograms for latest version

i have error of
```
[20:42:23 ERROR]: [DecentHolograms] Error initializing plugin 'DecentHolograms-2.8.4.jar' in folder 'plugins' (Is it up to date?)
java.lang.NoClassDefFoundError: space/arim/morepaperlib/MorePaperLib
        at eu.decentsoftware.holograms.api.DecentHologramsAPI.onLoad(DecentHologramsAPI.java:37) ~[DecentHolograms-2.8.4.jar:?]
        at eu.decentsoftware.holograms.plugin.DecentHologramsPlugin.onLoad(DecentHologramsPlugin.java:18) ~[DecentHolograms-2.8.4.jar:?]
        at io.papermc.paper.plugin.storage.ServerPluginProviderStorage.processProvided(ServerPluginProviderStorage.java:59) ~[folia-1.20.2.jar:git-Folia-"7aacf90"]
        at io.papermc.paper.plugin.storage.ServerPluginProviderStorage.processProvided(ServerPluginProviderStorage.java:18) ~[folia-1.20.2.jar:git-Folia-"7aacf90"]
        at io.papermc.paper.plugin.storage.SimpleProviderStorage.enter(SimpleProviderStorage.java:40) ~[folia-1.20.2.jar:git-Folia-"7aacf90"]
        at io.papermc.paper.plugin.entrypoint.LaunchEntryPointHandler.enter(LaunchEntryPointHandler.java:36) ~[folia-1.20.2.jar:git-Folia-"7aacf90"]
        at org.bukkit.craftbukkit.v1_20_R2.CraftServer.loadPlugins(CraftServer.java:520) ~[folia-1.20.2.jar:git-Folia-"7aacf90"]
        at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:273) ~[folia-1.20.2.jar:git-Folia-"7aacf90"]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1173) ~[folia-1.20.2.jar:git-Folia-"7aacf90"]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:343) ~[folia-1.20.2.jar:git-Folia-"7aacf90"]
        at java.lang.Thread.run(Thread.java:842) ~[?:?]
Caused by: java.lang.ClassNotFoundException: space.arim.morepaperlib.MorePaperLib
        at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:197) ~[folia-api-1.20.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:164) ~[folia-api-1.20.2-R0.1-SNAPSHOT.jar:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
        ... 11 more
```
after compile , what should i do?

**Comment by mani1232** [comment]
If you need Folia support, dm me in discord: mani123

**Comment by d0by1** [comment]
> If you need Folia support, dm me in discord: mani123

You can make a Pull Request to add the support to DecentHolograms. I would merge such a feature.

**Comment by mani1232** [comment]
#175 here

**Comment by Xyness** [comment]
Hi, any news?

**Comment by MightyPowerN** [comment]
Hello, is there any news in the future where this could be implemented? didn't want to create a new issue, since there isn't really an issue just a request. I don't want to be pushy I am just curious if in the future this will be planned I don't need an ETA or anything just asking around to see the future of this. 

Thank you for the contributors for doing a great job with this plugin and their time.

**Comment by d0by1** [comment]
Folia support has been added in commit de6c9f62f72baa5b84a103e78804a9ca16fa45f4.

Until the next official release is available, you can use the latest official development build from GitHub Actions:
https://github.com/DecentSoftware-eu/DecentHolograms/actions/runs/31413926787
